pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -189,6 +191,21 @@ func (r *PerformanceProfileReconciler) hypershiftReconcile(ctx context.Context,
 		return reconcile.Result{}, nil
 	}
 
+	machineconfigEncoded, err := encodeMachineConfig(convertMachineConfig(componentSet.MachineConfig))
+	if err != nil {
+		klog.Warningf("failed to Encode MachineConfig in ConfigMap %s: %v", instance.ObjectMeta.Name, err)
```

Comment:
again seems worthy to bubble up the error and set degraded conditions

Reply:
error part done. 
for degraded conditions see https://github.com/openshift/cluster-node-tuning-operator/pull/484#discussion_r1409195931